From d161ca700a2679c2827e6d93455d56f3fa2d14a2 Mon Sep 17 00:00:00 2001
From: Preeti Ahuja <preetia@codeaurora.org>
Date: Thu, 21 Aug 2014 16:26:24 -0700
Subject: [PATCH] SEEK: Add support for getAtr

Add support for RIL_REQUEST_SIM_GET_ATR request
for enabling getAtr api.

CRs-Fixed: 718469
Change-Id: I5fe8f8156b55e7feaa7bb2671cd8451cb8bd2f36
---
 include/telephony/ril.h | 20 ++++++++++++++++++++
 libril/ril.cpp          |  1 +
 libril/ril_commands.h   |  1 +
 3 files changed, 22 insertions(+)

diff --git a/include/telephony/ril.h b/include/telephony/ril.h
index bd3d2ed..224747b 100644
--- a/include/telephony/ril.h
+++ b/include/telephony/ril.h
@@ -4526,6 +4526,26 @@ typedef struct {
  */
 #define RIL_REQUEST_GET_ACTIVITY_INFO 135
 
+/**
+ * RIL_REQUEST_SIM_GET_ATR
+ *
+ * Get the ATR from SIM Card
+ *
+ * Only valid when radio state is "RADIO_STATE_ON"
+ *
+ * "data" is const int *
+ * ((const int *)data)[0] contains the slot index on the SIM from which ATR is requested.
+ *
+ * "response" is a const char * containing the ATR, See ETSI 102.221 8.1 and ISO/IEC 7816 3
+ *
+ * Valid errors:
+ *
+ * SUCCESS
+ * RADIO_NOT_AVAILABLE (radio resetting)
+ * GENERIC_FAILURE
+ */
+#define RIL_REQUEST_SIM_GET_ATR 136
+
 /***********************************************************************/
 
 
diff --git a/libril/ril.cpp b/libril/ril.cpp
index b1071ae..592f883 100755
--- a/libril/ril.cpp
+++ b/libril/ril.cpp
@@ -5167,6 +5167,7 @@ requestToString(int request) {
         case RIL_REQUEST_SIM_TRANSMIT_APDU_CHANNEL: return "SIM_TRANSMIT_APDU_CHANNEL";
         case RIL_REQUEST_GET_RADIO_CAPABILITY: return "RIL_REQUEST_GET_RADIO_CAPABILITY";
         case RIL_REQUEST_SET_RADIO_CAPABILITY: return "RIL_REQUEST_SET_RADIO_CAPABILITY";
+        case RIL_REQUEST_SIM_GET_ATR: return "SIM_GET_ATR";
         case RIL_REQUEST_SET_UICC_SUBSCRIPTION: return "SET_UICC_SUBSCRIPTION";
         case RIL_REQUEST_ALLOW_DATA: return "ALLOW_DATA";
         case RIL_REQUEST_GET_HARDWARE_CONFIG: return "GET_HARDWARE_CONFIG";
diff --git a/libril/ril_commands.h b/libril/ril_commands.h
index 163aa46..50e649a 100644
--- a/libril/ril_commands.h
+++ b/libril/ril_commands.h
@@ -150,3 +150,4 @@
     {RIL_REQUEST_STOP_LCE, dispatchVoid, responseLceStatus},
     {RIL_REQUEST_PULL_LCEDATA, dispatchVoid, responseLceData},
     {RIL_REQUEST_GET_ACTIVITY_INFO, dispatchVoid, responseActivityData},
+    {RIL_REQUEST_SIM_GET_ATR, dispatchInts, responseString},
-- 
2.6.2

